Assets/Scripts/HttpTopicClient.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.Networking;
+
+struct MessageContentType
+{
+    public const string Text = "text/plain";
+    public const string Binary = "application/octet-stream";
+}
+
+internal class Base64DecodedV1Token
+{
+    public string? api_key = null;
+    public string? endpoint = null;
+}
+
+public class HttpTopicSubscription
+{
+    private IEnumerator _poller;
+    private bool cancelled = false;
+
+    public bool Cancelled {
+        get { return cancelled; }
+    }
+
+    public void Cancel() {
+        Debug.Log("CANCELLING SUBSCRIPTION");
+        cancelled = true;
+    }
+
+    public HttpTopicSubscription() {}
+
+    public IEnumerator Poller
+    {
+        get { return _poller; }
+        set { _poller = value; }
+    }
+}
+
+public class HttpTopicClient
+{
+    private string endpoint;
+    private string authToken;
+
+    public HttpTopicClient(string authToken){
+        this.endpoint = GetEndpointFromKey(authToken);
+        Debug.Log("Using endpoint: " + this.endpoint);
+        this.authToken = authToken;
+    }
+
+    public void Publish(string cacheName, string topicName, byte[] message)
+    {
+        var messageString = System.Text.Encoding.UTF8.GetString(message);
```

Comment:
The docs for this very explicitly state that it will throw an exception for non-unicode data:

> Exceptions
> [ArgumentException](https://learn.microsoft.com/en-us/dotnet/api/system.argumentexception?view=net-8.0)
> 
> The byte array contains invalid Unicode code points.

https://learn.microsoft.com/en-us/dotnet/api/system.text.encoding.getstring?view=net-8.0#system-text-encoding-getstring(system-byte())

What am I missing?

Reply:
I ran a couple of tests in standard .NET, not Unity, and that didn't throw:
```
byte[] invalidUtf8Array = new byte[] { 0xFF, 0xFE, 0xFD };

string decodedString = System.Text.Encoding.UTF8.GetString(invalidUtf8Array);

UTF8Encoding utf8Throw = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
utf8Throw.GetString(invalidUtf8Array);
```
The default encoding returned "���". I could set up an encoding that threw, however.

Reply:
It looks like System.Text.Encoding.UTF8 is set up to return fallback characters for invalid UTF-8 characters: https://learn.microsoft.com/en-us/dotnet/api/system.text.encoding.utf8?view=net-8.0

Reply:
I'm just dealing in raw bytes in the client now and leaving all encoding concerns up to the user.